jbmc/src/java_bytecode/java_types.cpp

```diff
@@ -120,15 +120,24 @@ reference_typet java_array_type(const char subtype)
   return java_reference_type(symbol_type);
 }
 
-/// Return the type of the elements of a given java array type
-/// \param array_type The java array type
-/// \return The type of the elements of the array
-typet java_array_element_type(const symbol_typet &array_type)
+/// Return a const reference to the element type of a given java array type
+/// \param array_symbol The java array type
+const typet &java_array_element_type(const symbol_typet &array_symbol)
 {
   INVARIANT(
```

Comment:
@martin-cs may explain what the difference really is, but that looks rather like a `DATA_INVARIANT` to me...